src/niclips/image/_convert.py

```diff
@@ -62,53 +59,49 @@ def normalize(
     vmax: float | None = None,
 ) -> np.ndarray:
     """Normalize data using vmin/vmax if provided, otherwise data min/max."""
-    data = np.asarray(data)
-    if vmin is None:
-        vmin = np.nanmin(data)
-    if vmax is None:
-        vmax = np.nanmax(data)
-    data = (data - vmin) / max(vmax - vmin, EPS)
-    data = np.clip(data, 0, 1)
-    return data
+    vmin = np.nanmin(data) if vmin is None else vmin
+    vmax = np.nanmax(data) if vmax is None else vmax
+    scale = vmax - vmin
+    if scale > EPS:
+        return np.clip((data - vmin) / scale, 0, 1)
+    return np.zeros_like(data)
 
 
 def scale(
     img: Image.Image, height: int, resample: Image.Resampling | None = None
 ) -> Image.Image:
-    """Scale an image to a target height.
+    """Scale image isotropically to a target height.
 
-    Ensure width is even numbered.
+    Note: Ensure size is even numbered for video codec.
     """
-    height += height % 2
-    scale = height / img.height
-    width = int(scale * img.width) + (int(scale * img.width) % 2)
-    size = width, height
-    img = img.resize(size, resample=resample)
-    return img
+    # Currently rescale to isotropic to not stretch images in generated figures
+    # Can maintain aspect ratio by using image size (width, height)
+    # Leaving the code below to do so.
+    # scale = height / img.height
+    # width = int(img.width * scale) & ~1
+    height = height & ~1
+
+    return img.resize((height, height), resample=resample)
 
 
 def reorient(img: np.ndarray) -> np.ndarray:
     """Reorient image axes from XY (i.e. Nifti-like) to IJ (i.e. typical image-like)."""
     return np.flipud(np.swapaxes(img, 0, 1))
 
 
-def to_iso_ras(img: nib.nifti1.Nifti1Image) -> nib.nifti1.Nifti1Image:
+def to_ras(img: nib.nifti1.Nifti1Image) -> nib.nifti1.Nifti1Image:
     """Convert a nifti image to RAS orientation with isotropic resolution."""
     # Grab original filepath
     img_path = img.get_filename()
-    img = reorder_img(img, resample="nearest")
-    affine = img.affine
-    pixdim = np.diag(affine)[:3]
-    if not np.all(pixdim == pixdim[0]):
-        res = np.min(pixdim)
-        target_affine = affine.copy()
-        np.fill_diagonal(target_affine[:3, :3], res)
-
-        with warnings.catch_warnings():
-            warnings.simplefilter("ignore")
-            img = resample_img(
-                img, target_affine=target_affine, interpolation="nearest"
-            )
+    # Reorient to RAS
+    trgt_ornt = np.array([[0, 1], [1, 1], [2, 1]])
+    crnt_ornt = nib.orientations.io_orientation(img.affine)
+    xfm = nib.orientations.ornt_transform(crnt_ornt, trgt_ornt)
+    reoriented_affine = nib.orientations.inv_ornt_aff(xfm, img.shape)
+    new_affine = img.affine @ reoriented_affine
+    reoriented_data = nib.orientations.apply_orientation(get_fdata(img), xfm)
+    img = nib.Nifti1Image(dataobj=reoriented_data, affine=new_affine)
+
```

Comment:
I wonder whether this functionality is already implemented in `nibabel`. It feels like it should be. Maybe [`nibabel.funcs.as_closest_canonical`](https://nipy.org/nibabel/reference/nibabel.funcs.html#nibabel.funcs.as_closest_canonical)?

Reply:
I'll take a look at this - I was trying to find out where I had originally seen this code, but if we can find an equivalent method to call that's even better.

Reply:
Done in 201d44f